Sources/X509/RelativeDistinguishedName.swift

```diff
@@ -33,23 +34,36 @@ import SwiftASN1
 /// at index `i` does not guarantee it will remain at that location. As a result, ``RelativeDistinguishedName`` is
 /// not a `MutableCollection`.
 public struct RelativeDistinguishedName {
-    // TODO: Should we special-case this to the circumstance where we have only one attribute?
     @usableFromInline
-    var attributes: [Attribute]
-
+    var attributes: _TinyArray<Attribute>
+    
     /// Construct a ``RelativeDistinguishedName`` from a sequence of ``Attribute``.
     ///
     /// - Parameter attributes: The sequence of ``Attribute``s that make up the ``DistinguishedName``.
     @inlinable
-    public init<AttributeSequence: Sequence>(_ attributes: AttributeSequence) throws where AttributeSequence.Element == RelativeDistinguishedName.Attribute {
-        self.attributes = Array(attributes)
+    public init(_ attribute: Attribute) {
+        self.init(CollectionOfOne(attribute))
+    }
+    
+    /// Construct a ``RelativeDistinguishedName`` from a sequence of ``Attribute``.
+    ///
+    /// - Parameter attributes: The sequence of ``Attribute``s that make up the ``DistinguishedName``.
+    @inlinable
+    public init<AttributeSequence: Sequence>(_ attributes: AttributeSequence) where AttributeSequence.Element == RelativeDistinguishedName.Attribute {
+        self.attributes = .init(attributes)
+        Self._sortElements(&self.attributes)
+    }
+    
+    @inlinable
+    init(_ attributes: some Sequence<Result<RelativeDistinguishedName.Attribute, some Error>>) throws {
```

Comment:
Do we need this to be fully generic? Or can we just accept the actual type?

Reply:
Nope, have changed it: https://github.com/apple/swift-certificates/pull/101/commits/d2ca5723af35720530e6ef405a6a1f6743a77ce2